### PR TITLE
Fix sonatype issues

### PR DESCRIPTION
**Pull request by ddalcino**
This is an attempt to fix as many easily correctable issues discovered by sonatype as possible.

### CODE REVIEW

File: aqt/helper.py [hunk 1]
```diff
@@ -27,15 +27,16 @@
 import posixpath
 import secrets
 import sys
-import xml.etree.ElementTree as ElementTree
 from logging import getLogger
 from logging.handlers import QueueListener
 from pathlib import Path
 from typing import Callable, Dict, Generator, List, Optional, Tuple
 from urllib.parse import urlparse
+from xml.etree.ElementTree import Element
```
[the remark this answers is not in the document]
Comment (ddalcino) [reply aqt/helper.py:35]: @sonatype-lift ignore

Here, `Element` is used for type annotations. All parsing is done with defusedxml.

File: aqt/metadata.py [hunk 2]
```diff
@@ -22,12 +22,12 @@
...
 import operator
 import posixpath
-import random
 import re
+import secrets as random
 import shutil
 from logging import getLogger
 from typing import Callable, Dict, Generator, Iterable, Iterator, List, Optional, Tuple, Union
-from xml.etree import ElementTree as ElementTree
+from xml.etree.ElementTree import Element
```
[the remark this answers is not in the document]
Comment (ddalcino) [reply aqt/metadata.py:30]: @sonatype-lift ignore

Here, Element is used for type annotations. All parsing is done with defusedxml.

File: aqt/installer.py [hunk 3]
```diff
@@ -155,6 +155,7 @@ def _set_sevenzip(self, external):
                 [sevenzip, "--help"],
                 stdout=subprocess.DEVNULL,
                 stderr=subprocess.DEVNULL,
+                shell=True,
```
[the remark this answers is not in the document]
Comment (ddalcino) [reply aqt/installer.py]: Oops, I misunderstood the warning. Apparently, there are warnings for subprocess.run when shell=False *and* when shell=True.

**Comment by ddalcino** [comment]
I'm not sure how to fix the rest of the reported issues.

**Comment by miurahr** [comment]
Thanks. It may be easy to get wrong XML from attacked mirror site. We need to protect ourselves.

**Pull request merged**